engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/query/NadelFieldToService.kt

```diff
@@ -0,0 +1,54 @@
+package graphql.nadel.enginekt.transform.query
+
+import graphql.nadel.Service
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.util.copyWithChildren
+import graphql.nadel.schema.NadelDirectives
+import graphql.normalized.NormalizedField
+import graphql.normalized.NormalizedQuery
+import graphql.schema.GraphQLSchema
+
+class NadelFieldToService {
+
+    fun getServicesForTopLevelFields(query: NormalizedQuery, overallExecutionBlueprint: NadelOverallExecutionBlueprint): List<FieldAndService> {
+        return query.topLevelFields.flatMap { topLevelField ->
+            when {
+                NadelNamespacedFields.isNamespacedField(topLevelField, overallExecutionBlueprint.schema) -> {
```

Comment:
One upside of making `NadelOverallExecutionBlueprint` a global property is that you can use it in any function. We can start simplifying this code. This field -> pair(parent type, field def) -> service is a bit convoluted. Especially when repeated twice.

```kotlin
    private fun getService(field: NormalizedField): Service {
        val operationTypeName = field.objectTypeNames.single()
        val fieldCoordinates = makeFieldCoordinates(operationTypeName, field.name)
        return overallExecutionBlueprint.getService(fieldCoordinates)
            ?: error("Unable to field service for field at: $fieldCoordinates")
    }
```

And simply

```kotlin
  topLevelField.children.groupBy(::getField)
```

and

```kotlin
listOf(
  FieldAndService(
    field = topLevelField,
    service = getService(topLevelField),
  )
)
```

Reply:
Also note that instead of getting __"one"__ definition, we're ensuring there is __only one__ possible definition by using `objectTypeNames.single()`

Reply:
Sorry if this seems nitpicky but as soon as code starts getting too verbose it gets hard to read, and then hard to maintain. We really need to keep the complexity of the Nadel cosebase down so it doesn't become unmaintainable again.

It's well worth creating helper functions as soon as code becomes too much to read and doesn't tell a story when reading it.